### PR TITLE
refactor: Store presentation text content in Postgres

### PR DESCRIPTION
**Pull request by paultrudel**
### What does this PR do?

Sends the text content of each presentation from `bbb-web` to `akka-apps` so that it can be stored in Postgres.

### Motivation

Currently the BBB client parses a presentation's text content from the `txtUri` that is associated with each page in the presentation. Instead of doing this `bbb-web` now directly sends the text content of each page to `akka-apps` to be stored in Postgres so that it can be accessed directly by the client.

### CODE REVIEW

**Comment by gustavotrott** [comment]
Hey, I'm facing problems with special chars, can you take a look?

[test.docx](https://github.com/bigbluebutton/bigbluebutton/files/13029773/test.docx)

image

image

**Comment by paultrudel** [comment]
> Hey, I'm facing problems with special chars, can you take a look?
> 
> [test.docx](https://github.com/bigbluebutton/bigbluebutton/files/13029773/test.docx)
> 
> image
> 
> image

That is strange. I am not facing that issue when I upload the file.
image

I will continue to investigate and try to recreate the error.

**Comment by paultrudel** [comment]
I created a file with a bunch of special characters. I did not run into any issues with parsing them.
[special-char-test.odt](https://github.com/bigbluebutton/bigbluebutton/files/13032643/special-char-test.odt)
image

**Comment by gustavotrott** [comment]
> I created a file with a bunch of special characters. I did not run into any issues with parsing them. [special-char-test.odt](https://github.com/bigbluebutton/bigbluebutton/files/13032643/special-char-test.odt) image

Yeah, it's probably related to the default `$LANG` of the OS.
I'll force to read as UTF-8 and test here, then I send a suggestion to your code.
Thanks a lot for checking it out!

File: bbb-common-web/src/main/scala/org/bigbluebutton/api2/MsgBuilder.scala [hunk 1]
```diff
@@ -98,10 +99,15 @@ object MsgBuilder {
       val width = w.toDouble
       val height = h.toDouble
 
+      val contentUrl = new URL(txtUrl)
+      val reader = new BufferedReader(new InputStreamReader(contentUrl.openStream()))
```
Comment (gustavotrott) [inline bbb-common-web/src/main/scala/org/bigbluebutton/api2/MsgBuilder.scala:103]:
```suggestion
      val reader = new BufferedReader(new InputStreamReader(contentUrl.openStream(), StandardCharsets.UTF_8))
```

File: bbb-common-web/src/main/scala/org/bigbluebutton/api2/MsgBuilder.scala [hunk 2]
```diff
@@ -6,8 +6,9 @@ import org.bigbluebutton.common2.domain.{ DefaultProps, PageVO, PresentationPage
 import org.bigbluebutton.common2.msgs._
 import org.bigbluebutton.presentation.messages._
 
-import java.io.IOException
+import java.io.{ BufferedReader, IOException, InputStreamReader }
 import java.net.URL
+import java.util.stream.Collectors
```
Comment (gustavotrott) [inline bbb-common-web/src/main/scala/org/bigbluebutton/api2/MsgBuilder.scala:11]:
```suggestion
import java.nio.charset.StandardCharsets
import java.util.stream.Collectors
```

**Review by gustavotrott — CHANGES_REQUESTED** [review]
Force reading as UTF-8.

**Review by gustavotrott — APPROVED** [review]
LGTM!

**Pull request merged**